addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -695,7 +702,7 @@ export class OdooEditor extends EventTarget {
         if (!this._historyStepsActive) {
             return;
         }
-        this.observerFlush();
+        this.sanitize();
```

Comment:
> I was questioning the fact we now sanitize during `historyStep`, that seems strange to me considering not other sanitization call seems to be removed in this PR and especially without explanations for the reader to understand :shrug: 

The main problem was that there was not sanitization when an automatic step was made (see `this.observerTimeout`).
Which made me think: when do we want to make an historyStep without sanitizing? Maybe when we know that nothing is to be sanitized like in undo/redo. But as the sanitization is not performance heavy, we can simply sanitize each time in order to avoid forgetting to sanitize before an historyStep.

> So I went and have a look where sanitize is currently called, I see this:
> ![image](https://user-images.githubusercontent.com/10338094/170277416-27aadcb5-f94f-473b-9b9b-d1fe156373cf.png)
> 
> So apparently when reaching that code, we sanitize twice? shrug

The double sanitization was a mistake, I've removed the `this.sanitize();this.historystep();`.

> Clearly there is more to think about here and comments to explain what's going on, that's not supposed to reach the review step. 

What is it supposed to reach? Isn't it the role of the reviewer to spot problem that the developer didn't foresee?

> Even your commit message (mentioning the wrong apps) is clearly too small considering all the discussions that went around this problem we are trying to solve.

Indeed, there is one module mistake in the commit, thank you for spotting it.
For me, the discussion is simple: there was a behavior before the new editor that we broke and that need a fix otherwise the part of the code that require that behavior does not work. If we had fixed it in the editor merge, no discussions would have ever been referenced.

> Probably also more things to consider as explained here [#85227 (comment)](https://github.com/odoo/odoo/pull/85227#discussion_r880401952) (at least as comment addition but probably also unifying the behavior of `o_not_editable` and `getReadonlyAreas`?)
I'm suggesting to revert the commit of #85227 in master as you do not want to do it in 15.0. (I've tested and the fix #85227 became useless as some point as I can't reproduce it when reverting it in 15.0 (without this PR)).

Reply:
> The double sanitization was a mistake, I've removed the this.sanitize();this.historystep();.

 >>   Clearly there is more to think about here and comments to explain what's going on, that's not supposed to reach the review step.

> What is it supposed to reach? Isn't it the role of the reviewer to spot problem that the developer didn't foresee?

Exactly: spot problems that the developer did not foresee. I don't want to be mean but there was nothing to "foresee" here :grimacing:  You add a `sanitize` in `historyStep`:
- You thus grep `sanitize` to see if any useless call now remains -> yes the one next to a call to `historyStep`
- You thus grep `historyStep` to see if you actually wanted to sanitize each time it is called -> you also find the `sanitize` calll next to `historyStep`

-> You forgot to do this during development? Not a problem, you then self-review your PR, see a `sanitize` call added without comment and not any other call removed, you go back to your code to check.

-> You missed that during your self-review? Not a problem either, it's a mistake, everyone make some but, as I said, that's not my role as a reviewer to spot **that kind** of mistake (more often that not, **I will miss those and they will get merged**).

Again, I really don't want to be mean. You apparently did not like me saying this should not reach the review step, but that was not about the `sanitize` call only. It was paired with the unwritten map I mentioned below https://user-images.githubusercontent.com/10338094/170280519-ca8fa1a2-1585-4861-8b07-314bdb20d6b7.png, the huge lack of comments in both the code and the commit message, ... I really got a real vibe of "Ok, it should work, let's send to QSM, maybe he will r+"... :weary: 

Reviewing this kind of PR (spotting that kind of issues) takes more time it would take me to do the actual development:
- "Ok so a call to `sanitize` there... why...? Don't know... let's grep, mmh there are useless calls here and there"
- "Ok so a map... why and how is it used... let's add a debugger to try and understand"
- etc

> If we had fixed it in the editor merge, no discussions would have ever been referenced.

Are you actually saying that if we developed without bugs, we would not need bug fixes and thus we don't need to explain bug fixes? :s 

> I'm suggesting to revert the commit of https://github.com/odoo/odoo/pull/85227 in master [...]

From my understanding, @dmo-odoo wanted to keep the functionality and as I explained I have nothing against it but it would require unifications in behaviors IMO. I let you decide, I am fine closing my eyes until a bug is reported if you prefer (as a bug is actually unlikely I guess, it's more about having robust behaviors).

> [...] as you do not want to do it in 15.0.

More the stable policy than me :wink: If you judge the code is recent enough to break stable policy, revert in stable and r+, I am fine closing my eyes too, that's just not something I would do myself.